syntax/scan.go

```diff
@@ -22,7 +22,7 @@ import (
 type Token int8
 
 const (
-	ILLEGAL Token = iota
+	ILLEGAL Token = iota // deprecated: illegal tokens are at the bottom of this block
```

Comment:
It's not deprecated: it's still the name for zero value of Token, which indicates some kind of mistake. So let's remove this comment.